influxdata/platform
# new duration literals

Closes #558 

_Briefly describe your proposed changes:_
_What was the problem?_

The Flux grammar did not support day and week duration literals.

_What was the solution?_

I defined new tokens for `d` and `w` in a duration literal in the Flux grammar file. I also delayed converting an abstract duration literal type to a concrete duration Go type. That conversion now takes place during the construction of a semantic graph.

To test:
* rebuild the project by running `make` from the project root
* `cd` to the bin/darwin directory
* start the Flux cli via `./influx repl`
* Enter the new duration literals to see that they are correctly parsed and interpreted

    ```
    >1d
    >24h0m0s
    >
    >1w
    >168h0m0s
    ```

  - [ ] swagger.json updated (if modified Go structs or API)
  - [ ] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
